### PR TITLE
Upload logs to fedora

### PR DESCRIPTION
**Pull request by spetrosi**

### CODE REVIEW

File: tests/general/test.sh [hunk 1]
```diff
@@ -206,11 +207,37 @@ all:
...
 rolesIsVirtual() {
     # Returns 0 if provisioned with "how: virtual"
-    local tmt_tree_provision=$1
     grep -q 'how: virtual' "$tmt_tree_provision"/step.yaml
     echo $?
 }
 
+rolesUploadLogs() {
+    local logfile=$1
+    if [ -z "$LINUXSYSTEMROLES_SSH_KEY" ]; then
```
Comment (richm) [inline tests/general/test.sh:216]: Is `LINUXSYSTEMROLES_SSH_KEY` set as a github action secret?
Is there any way that this value could be echoed to the output?  For example, in github actions you can run a workflow in debug mode which is equivalent to using bash `set -x` - in a github action `run` block, I believe they provide some special logic which prevents the value of secrets referenced as environment variables from being output, or they somehow obscure the value.  I'm not sure if something similar exists for these tmt scripts.

Is there any guidance from the tmt team about how to handle secret values such as this in external workflows?

Comment (spetrosi) [reply tests/general/test.sh:216]: See this test workflow: https://github.com/spetrosi/network/actions/runs/9780735088/job/27003112712
In spetrosi/network, I added new secrets added to workflow in https://github.com/spetrosi/network/pull/50/files
Github hides secrets in the output like this:
`variables: ANSIBLE_VER=2.16;REPO_NAME=network;PR_NUM=61;ARTIFACTS_DIR=/srv/pub/alt/***/logs/tf-network_61-20240703-153029/artifacts;ARTIFACTS_URL=https://dl.fedoraproject.org/pub/alt/***/logs/tf-network_61-20240703-153029/artifacts;LINUXSYSTEMROLES_USER=***;LINUXSYSTEMROLES_DOMAIN=***;LINUXSYSTEMROLES_SSH_KEY=  ***`

I'll try running with the debug mode too and check if GH leaks something in this case.

One risk here is Github Action https://github.com/myrotvorets/set-commit-status-action that I use to set check status for commits, it requires providing `token: ${{ secrets.GITHUB_TOKEN }}`. Info about tokens in https://docs.github.com/en/actions/security-guides/automatic-token-authentication, it is set per workflow and per repository, the given workflow defines the following permissions:
```
permissions:
  contents: read
  # This is required for the ability to create/update the Pull request status
  statuses: write
```
So leaking it does not seem like a risk.

Comment (spetrosi) [reply tests/general/test.sh:216]: The run itself failed because in the test git forgot it's `-C` option, I am troubleshooting.

**Comment by spetrosi** [comment]
Last successful tests are in https://github.com/spetrosi/network/pull/64

**Comment by spetrosi** [comment]
@richm I am going on a 10-day PTO, so let me merge this now. I am testing this on a PR on my network fork https://github.com/spetrosi/network/pull/64. The workflow is at https://github.com/spetrosi/network/blob/main/.github/workflows/tmt.yml and test plan is within the repo in https://github.com/spetrosi/network/blob/main/plans/general.fmf.
Workflow runs from the main branch, so changes to workflow do not work until you merge them.
Storing test plan within the repo 

This functionality simplifies reaching logs, users click **details** from the checks. So we can use this when we move to public ranch too.

**Pull request merged**